frame/staking/src/lib.rs

```diff
@@ -761,6 +766,105 @@ impl<AccountId, Balance: HasCompact + Zero> UnappliedSlash<AccountId, Balance> {
 	}
 }
 
+/// Something that defines the maximum number of nominations per nominator based on a curve.
+pub trait NominationsQuota<Balance> {
+	/// Maximum number of nominations. The method `curve` implements the nomination quota curve and
+	/// should not be used directly. However, `get_quota` returns the bounded maximum number of
+	// nominations based on `fn curve` and the nominator's balance.
+	type MaxNominations: Get<u32>;
+
+	/// Returns the voter's nomination quota within reasonable bounds [`min`, `max`], where `min`
+	/// is 1 and `max` is `Self::MaxNominations`.
+	fn get_quota(balance: Balance) -> u32 {
+		Self::curve(balance).clamp(1, Self::MaxNominations::get())
+	}
+
+	// Returns the voter's nomination quota based on its balance and a curve.
+	fn curve(balance: Balance) -> u32;
+}
+
+/// A nomination quota that allows up to MAX nominations for all validators.
+pub struct FixedNominationsQuota<const MAX: u32>;
+impl<Balance, const MAX: u32> NominationsQuota<Balance> for FixedNominationsQuota<MAX> {
+	type MaxNominations = ConstU32<MAX>;
+
+	fn curve(_: Balance) -> u32 {
+		MAX
+	}
+}
+
+/// A static tracker for the election data snapshot.
+///
+/// Computes the (SCALE) encoded byte length of a snapshot based on static rules, without any actual
```

Comment:
```suggestion
impl<Balance, T: Get<u32>> NominationsQuota<Balance> for T {
	type MaxNominations: T;
	fn curve(_) -> u32 { T::get() }
}
```

Mostly sharing it because it is a nice trait gymnastics, but it is probably not as readable and clear `FixedNominationsQuota` having an explicit name. Up to you. We could also have both.